regparser/web/jobs/views.py

```diff
@@ -148,3 +194,172 @@ def retrieve(self, request, *args, **kwargs):
+        Overrides the method from ``BaseViewList`` in order to pass the
+        arguments appropriate for the ``pipeline`` command.
+
+        Side Effects
+            Runs the ``pipeline`` command.
+
+        :arg dict validated_data: Incoming data from the POST that's already
+        been validated by the serializer.
+
+        :rtype: list[str]
+        :returns: The components of the argument string in list form.
+        """
+        return [
+            "pipeline",
+            str(validated_data["cfr_title"]),
+            str(validated_data["cfr_part"]),
+            eregs_site_api_url,
+        ]
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class PipelineJobViewInstance(JobViewInstance):
+    queryset = PipelineJob.objects.all()
+    serializer_class = PipelineJobSerializer
+    sub_path = "pipeline/"
+
+
+class ProposalPipelineJobViewList(BaseViewList, JobViewList):
+    queryset = ProposalPipelineJob.objects.all()
+    serializer_class = ProposalPipelineJobSerializer
+    sub_path = "proposal-pipeline/"
+
+    def build_eregs_args(self, validated_data):
+        """
+        Overrides the method from ``BaseViewList`` in order to pass the
+        arguments appropriate for the ``proposal_pipeline`` command.
+
+        Impure
+            Reads the contents of the proposal file from the filesystem (in
+            future, the DB, but impure either way).
+        Side Effects
+            Runs the ``proposal_pipeline`` command.
+
+        :arg dict validated_data: Incoming data from the POST that's already
+        been validated by the serializer.
+
+        :rtype: list[str]
+        :returns: The components of the argument string in list form.
+        """
+        reg_file = RegulationFile.objects.get(
+            hexhash=validated_data["file_hexhash"])
+        # TODO: This is a total hack; we should not be storing the contents in
+        # the DB but reading the file from the filesystem. Only doing this
+        # temporarily before changing the proposal_pipeline command to work
+        # differently.
+        path = reg_file.file.storage.path(reg_file.file.name)
+        return [
+            "proposal_pipeline",
+            path,
+            eregs_site_api_url
+        ]
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class ProposalPipelineJobViewInstance(JobViewInstance):
+    queryset = ProposalPipelineJob.objects.all()
+    serializer_class = ProposalPipelineJobSerializer
+    sub_path = "proposal-pipeline/"
+
+
+class FileUploadView(mixins.ListModelMixin, mixins.CreateModelMixin,
+                     generics.GenericAPIView):
+    parser_classes = (FileUploadParser, MultiPartParser)
+    parser_classes = (MultiPartParser,)
+    serializer_class = FileUploadSerializer
+    queryset = RegulationFile.objects.all()
+    lookup_field = "hexhash"
+    size_limit = 100000000  # Arbitrary 100MB limit.
```

Comment:
Probably! I also think it's still an open question as to whether the extra overhead incurred by loading the whole thing into memory to hash it (or to write it into the DB, for that matter) is worth it for the benefits of rejecting duplicate files early.